test/support.js

```diff
@@ -30,9 +30,6 @@ beforeEach(function() {
       {type: 'STUB'}
     ]
   });
-
-  // auto attach data sources to models
-  loopback.autoAttach();
```

Comment:
All these three comments are related to the first commit https://github.com/strongloop/loopback/commit/699bc7aa974c4a46cd8b6d2da71d5088ad532fb6 which is changing the test setup, removing `autoAttach` from a global beforeEach hook and enabling it only in the test suites that depend on that.

The reason why I need the change is that it's the fastest way how to reduce the amount of crazy cross-test side effect caused the global shared registry. I spend about one or two hours investigating the problem described below and I didn't want to waste more time on that.

Here is the quiz for you. Look at the following code and tell me, why the type of the property `ServerToken.userId` was changed from `String` to `Number` between the line 2 and line 3?

```js
before(setupServer);
before(setupClient);
beforeEach(seedServerData);
beforeEach(seedClientData);
```

.
.
.
.
.
.
.
.
.

Hint: the global beforeEach hook calling `autoAttach` is executed between the line 2 and 3. 

.
.
.
.

Do you know the answer yet? ;-)

.
.
.
.
.
.


`autoAttach` attached `User` and `AccessToken` models to a in-memory datasource. When these models are attached, model relations are resolved and `AccessToken.userId` is defined as a `Number` property, because `User.id` is a Number property injected by the in-memory datasource.

Unfortunately, it seem like the action has also a side effect of modifying the type of `userId` property in the descendant model `ServerToken`, which is related to `ServerUser` (and `ServerUser.id` has the id as a `String`).

Again, I really didn't want to spend any more time investigating it. Considering that we want to remove both autoAttach in the future, this small change is IMO going in the right direction, as it's reducing the number of tests depending on autoAttach.

However, if you feel I should debug the underlying issue and fix the probable bug in the juggler instead, then I can do that too.

/cc @raymondfeng